File: src/main/java/dominio/metodologias/OperacionRelacional.java
package dominio.metodologias;

public enum OperacionRelacional {
	
	Mayor{
		public boolean aplicarA(int num1, int num2) {
			return num1 > num2;
		}
		public String toString(){
			return ">";
		}
	},
	
	Menor{
		public boolean aplicarA(int num1, int num2) {
			return num1 < num2;
		}
		public String toString(){
			return "<";
		}
	},
	
	Igual{
		public boolean aplicarA(int num1, int num2) {
			return num1 == num2;
		}
		public String toString(){
			return "=";
		}
	};
	
	public abstract boolean aplicarA(int num1, int num2);
	public abstract String toString();

}

File: src/main/java/dondeInvierto/dominio/metodologias/OperacionRelacional.kt
package dondeInvierto.dominio.metodologias

enum class OperacionRelacional {

    Mayor {
        override fun aplicarA(num1: Int, num2: Int) = num1 > num2

        override fun toString() = ">"
    },

    Menor {
        override fun aplicarA(num1: Int, num2: Int) = num1 < num2

        override fun toString() = "<"
    },

    Igual {
        override fun aplicarA(num1: Int, num2: Int) = num1 == num2

        override fun toString() = "="
    };

    abstract fun aplicarA(num1: Int, num2: Int): Boolean
    abstract override fun toString(): String

}
